app/controllers/manageBusinesses/cease/ViewAllCeasedBusinessesController.scala

```diff
@@ -16,34 +16,66 @@
-import config.{AgentItvcErrorHandler, FrontendAppConfig}
+import config.{AgentItvcErrorHandler, FrontendAppConfig, ItvcErrorHandler}
 import controllers.agent.predicates.ClientConfirmedController
-import controllers.predicates.SessionTimeoutPredicate
-import play.api.i18n.I18nSupport
-import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
-import utils.AuthenticatorPredicate
+import enums.JourneyType.Manage
+import models.incomeSourceDetails.IncomeSourceDetailsModel
+import play.api.Logger
+import play.api.mvc._
+import services.{IncomeSourceDetailsService, SessionService}
+import uk.gov.hmrc.auth.core.AuthorisedFunctions
+import utils.{AuthenticatorPredicate, IncomeSourcesUtils}
+import views.html.manageBusinesses.cease.ViewAllCeasedBusinesses
 
-import javax.inject.Inject
-import scala.concurrent.ExecutionContext
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
 
+@Singleton
+class ViewAllCeasedBusinessesController @Inject()(val viewAllCeasedBusinesses: ViewAllCeasedBusinesses,
+                                                  val authorisedFunctions: AuthorisedFunctions,
+                                                  val incomeSourceDetailsService: IncomeSourceDetailsService,
+                                                  val sessionService: SessionService,
+                                                  val auth: AuthenticatorPredicate)
+                                                 (implicit val ec: ExecutionContext,
+                                             implicit override val mcc: MessagesControllerComponents,
+                                             implicit val itvcErrorHandler: ItvcErrorHandler,
+                                             implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                             val appConfig: FrontendAppConfig) extends ClientConfirmedController
+  with FeatureSwitching with IncomeSourcesUtils {
 
-//THIS CONTROLLER IS A PLACEHOLDER FOR THE PAGE TO BE BUILT IN MISUV-7077 (new income sources landing page)
-class ViewAllCeasedBusinessesController @Inject()(auth: AuthenticatorPredicate,
-                                                  val authorisedFunctions: FrontendAuthorisedFunctions,
-                                                  val checkSessionTimeout: SessionTimeoutPredicate)
-                                                 (implicit val appConfig: FrontendAppConfig,
-                                             mcc: MessagesControllerComponents,
-                                             val ec: ExecutionContext,
-                                             val itvcErrorHandlerAgent: AgentItvcErrorHandler)
-  extends ClientConfirmedController with FeatureSwitching with I18nSupport {
-
-  def show(): Action[AnyContent] = Action {
-    Ok("")
+  def show(isAgent: Boolean): Action[AnyContent] = auth.authenticatedAction(isAgent) { implicit user =>
```

Comment:
:100: - I think we should start using this approach right away (drop using showAgent method) and use joint method as removal of isAgent flag is on the horizon;